### PR TITLE
Fix ctrl_match with _entry_separator

### PR DESCRIPTION
**Pull request by blindFS**
This PR fixes some errors of `ctrl_match` because of non-explicit separator in match body.

image

### CODE REVIEW

File: src/scanner.c
Comment (mrdgo) [inline src/scanner.c:1]: hmmm

Comment (fdncred) [reply src/scanner.c:1]: do you mean to have scanner.s included in this PR?

Comment (blindFS) [reply src/scanner.c:1]: It's an accident, forgot to remove it from the branch...

**Comment by fdncred** [comment]
Thanks

**Pull request merged**